packages/deployment/lib/kes.js

```diff
@@ -179,9 +196,260 @@ class UpdatedKes extends Kes {
       dest
     ).then(() => {
       this.Lambda.messageAdapterZipFileHash = new this.Lambda(this.config).getHash(src);
-    }).then(() => super.compileCF());
+      return this.superCompileCF();
+    });
+  }
+
+  /**
+   * Modified version of Kes superclass compileCF method
+   *
+   * Compiles a CloudFormation template in Yaml format.
+   *
+   * Reads the configuration yaml from `.kes/config.yml`.
+   *
+   * Writes the template to `.kes/cloudformation.yml`.
+   *
+   * Uses `.kes/cloudformation.template.yml` as the base template
+   * for generating the final CF template.
+   *
+   * @returns {Promise} returns the promise of an AWS response object
+   */
+  async superCompileCF() {
+    const lambda = new this.Lambda(this.config);
+    if (this.config.lambdaProcess) {
+      this.config = await lambda.process();
+    }
+    else {
+      lambda.buildAllLambdaConfiguration();
+    }
+
+    let cf;
+
+    // Inject Lambda Alias values into configuration,
+    // then update configured workflow lambda references
+    // to reference the generated alias values
+    if (this.config.useWorkflowLambdaVersions === true) {
```

Comment:
To make sure I'm understanding this variable name correctly: we're only versioning lambdas used as tasks in step functions?

Reply:
Yes, that's correct.   API lambdas that aren't intended to be used in step functions are ignored.